packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx

```diff
@@ -23,23 +23,11 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
 
   // setup initial threads
   useEffect(() => {
-    // set pinned threads
-    setThreads(
-      app.threads.listingStore.getThreads({
-        topicName,
-        stageName,
-        pinned: true,
-      })
-    );
-    app.threads.loadNextPage({ topicName, stageName }).then(() => {
-      // add unpinned threads
-      setThreads((oldThreads) => [
-        ...oldThreads,
-        ...app.threads.listingStore.getThreads({
-          topicName,
-          stageName,
-          pinned: false,
-        }),
+    app.threads.loadNextPage({ topicName, stageName, includePinnedThreads: true }).then((t) => {
+      // Fetch first 20 + unpinned threads
+      setThreads([
+        ...threads,
+        ...t,
```

Comment:
I am getting error here and infinite loader sometimes when I go to General => Overall and back to General.

![image](https://user-images.githubusercontent.com/14819225/225892458-ed7b4dc7-d2b9-4423-aebc-13f8dd0192f6.png)

Reply:
Also, is this suppose to be running only once on component load? I assume yes, so `...threads,` is not necessary as it is always empty arrays? Also, make sure that dependency array for this useEffect is filled out correctly. eslint helps with that
![image](https://user-images.githubusercontent.com/14819225/225893559-b4bef55f-fd05-4b7e-bad7-4f51f3ee973b.png)